tools/VASP2JSON.py

```diff
@@ -0,0 +1,189 @@
+    """
+
+    jsonfile = open(jsonfilename, "w")
+    # print >>jsonfile, "# Header\n"
+    # print("# Header", file=jsonfilename)
+    allData = [data]
+    allDataHeader = {}
+    allDataHeader["EnergyStyle"] = "electronvolt"
+    allDataHeader["StressStyle"] = "kB"
+    allDataHeader["AtomTypeStyle"] = "chemicalsymbol"
+    allDataHeader["PositionsStyle"] = "angstrom"
+    allDataHeader["ForcesStyle"] = "electronvoltperangstrom"
+    allDataHeader["LatticeStyle"] = "angstrom"
+    allDataHeader["Data"] = allData
+
+    myDataset = {}
+
+    myDataset["Dataset"] = allDataHeader
+    sjson = json.dump(myDataset, jsonfile, indent=2, sort_keys=True)
+    return
+
+
+def print_num_atoms_per_type(myPOSCARFile):
+    """
+    print_num_atoms_per_type takes the POSCAR specified by the user and parses it
+    for the number of each atom types, which is assumed to be
+    in lines 7 of the POSCAR file (so if that's not the case this will need
+    to be modified in numberAtomsLine). This function then returns
+    a the number of the atom type for each indvidual atoms for each configuration.
+    Line 6 in the POSCAR is not actually used by VASP, and the ordering of the
+    atom types is determined by the POTCAR file, and listed in the OUTCAR file.
+
+    :param myPOSCARFile: str, filename of POSTCAR
+    :return: number of atoms of each type/ List[Int]
+    """
+    with open(myPOSCARFile, "rt") as f1:
+        lines = f1.readlines()
+    numberAtomsLine = lines[6]
+    columnsNumberAtom = numberAtomsLine.split()
+    num_atoms_per_type = [int(num) for num in columnsNumberAtom]
+    return num_atoms_per_type
+
+
+num_atoms_per_type = print_num_atoms_per_type(POSCAR_file)
+num_atom_types = len(num_atoms_per_type)
+order_atom_types = []
+listAtomTypes = []
+
+# Start parsing through OUTCAR looking for keywords that are assocaited with the
+# different values for the data needed, such as forces or positions
+
+with open(OUTCAR_file, "rt") as f2:
+    lines = f2.readlines()
+outcar_config_number = 1
+
+for i in range(0, len(lines)):
+    line = lines[i]
+    # Look for the ordering of the atom types
+    # can grab atom labels from VRHFIN lines at top of OUTCAR
+    # (These will only show up once for each element at in the OUTCAR)
+    if "VRHFIN" in line:
+        order_atom_types.append(line.split()[1][1:].strip(" :"))
+        if len(order_atom_types) == num_atom_types:
+            for atom_type in range(0, num_atom_types):
+                totalAtoms = int(num_atoms_per_type[atom_type])
+                atomType = str(order_atom_types[atom_type])
+                for n in range(0, totalAtoms):
+                    listAtomTypes.append(atomType)
+    # Look for number of atoms in configuration
+    if "number of ions" in line:
+        columns = line.split()
+        natoms = int(columns[11])
+    # Look for lattice vectors for configuration
+    if "direct lattice vectors" in line:
+        line = lines[i + 1]
+        columns = line.split()
```

Comment:
```
x1, x2, x3 = [float(x) for x in line[i + 1].split()]
```
or we can set it just to a single var
```
lattice_x = [float(x) for x in line[i + 1].split()]
```
or we could get the entire lattice in one
```
lattice = [[float(x) for x in line[i + j].split()] for j in range(3)]
```